futures/src/client.rs

```diff
@@ -52,50 +52,16 @@ impl Default for ConnectionOptions {
 pub type ConnectionConfiguration = lapin_async::connection::Configuration;
 
 /// A heartbeat task.
-pub struct Heartbeat {
+pub struct Heartbeat<PF> {
     handle: Option<HeartbeatHandle>,
-    pulse: Box<Future<Item = (), Error = io::Error> + Send>,
+    pulse: PF,
 }
 
-impl Heartbeat {
```

Comment:
Why are you reverting this too?

Reply:
That was not a revert, it's rather a refactoring. It is my attempt to avoid `Box`-ing the `pulse` field. 

If you can find out a better solution that would be great.

Reply:
I'd prefer to leave it as is actually